toolkit.nu

```diff
@@ -0,0 +1,66 @@
+# this module regroups a bunch of development tools to make the development
+# process easier for anyone.
+#
+# the main purpose of `toolkit` is to offer an easy to use interface for the
+# developer during a PR cycle.
+
+
+# check that all the tests pass
+export def test [
+] {
+    print "toolkit test: not implemented!"
+}
+
+# run all the necessary checks and tests to submit a perfect PR
+export def "check pr" [
+] {
+    generate-file-list
+    test
+}
+
+export def main [] { help toolkit }
+
+export def generate-file-list [ --full ] {
+    let start = "let files = ["
+
+    mut files = [""]
+
+    if $full {
+        # all the *.nu files in the repo
+        # exept for `before_v0.60`
+        # do nothing.. as $files has already all files
+        print "checking all files..."
+        mut $files = glob **/*.nu --exclude [before_v0.60/**]
+    } else {
+        $files = (git diff --name-only origin/main | lines | each { path expand })
```

Comment:
Should we filter here for `.nu` files as well?

Reply:
Yup, that is the problem I have right now!
I'll comment down below